lib/chef/dsl/secret.rb

```diff
@@ -30,16 +30,19 @@ module Secret
       #                         perform the secret lookup
       # @option config [Hash] The configuration that the named service expects
       #
+      # @return result [Object] The response object type is determined by the fetcher. See fetcher documentation
```

Comment:
It feels like this is going to get us in trouble. Should we instead just define our own object for a more consistent return type even if that varies somewhat between secrets systems.

Reply:
I'll look across secrets systems a bit today.  So far, I've seen where we can return a string, a binary (really just a string, hex-encoded), and possibly a hash.

Reply:
Is there any expectation that a user should be able to transition between secrets systems without having to modify their cookbooks?

Reply:
No, it is not a goal of this work.  Each of the systems has different quirks, and it's not expected that switching between them will be seamless.

Reply:
It would seem like this makes it *easier* to migrate but it can't possible fully abstract away the differences.